x-pack/plugins/ecs_data_quality_dashboard/server/routes/results/post_results.ts

```diff
@@ -51,20 +46,20 @@ export const postResultsRoute = (
         }
 
         try {
-          // Confirm user has authorization for the pattern payload
-          const { pattern } = request.body.rollup;
+          // Confirm user has authorization for the indexName payload
+          const { indexName } = request.body;
           const userEsClient = services.core.elasticsearch.client.asCurrentUser;
           const privileges = await userEsClient.security.hasPrivileges({
-            index: [{ names: [pattern], privileges: ['all', 'read', 'view_index_metadata'] }],
+            index: [{ names: [indexName], privileges: ['all', 'read', 'view_index_metadata'] }],
```

Comment:
Consider revising the implementation of this `privileges` check, because the following privileges are required to check an index:

- `monitor` or `manage`
- `view_index_metadata`
- `read`

As illustrated by the screenshot below:

![data_quality_with_has_viewer_role](https://github.com/elastic/kibana/assets/4459398/3cfc3e27-d25d-453b-aca2-88b41ae9b66a)

Consider:
- The `all` privilege is not required
- _either_ `monitor` or `manage` is required to check an index
  - In a future release, `read` may be expanded to be `read` or `read_cross_cluster`, if the Elasticsearch `stats` and `_ilm/explain` APIs are enhanced to support CCS.

Reply:
> Consider revising the implementation of this `privileges` check, because the following privileges are required to check an index:
> 
> * `monitor` or `manage`
> * `view_index_metadata`
> * `read`
> 
> As illustrated by the screenshot below:
> 
> ![data_quality_with_has_viewer_role](https://private-user-images.githubusercontent.com/4459398/298719371-3cfc3e27-d25d-453b-aca2-88b41ae9b66a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY2MjIwNTYsIm5iZiI6MTcwNjYyMTc1NiwicGF0aCI6Ii80NDU5Mzk4LzI5ODcxOTM3MS0zY2ZjM2UyNy1kMjVkLTQ1M2ItYWNhMi04OGI0MWFlOWI2NmEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMzBUMTMzNTU2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTMwNWZlNDM0ZTFlNWM3ODA4ZWU5YWQ0Y2VjYjA0MjM3MWMzZGUyYTk2ODY1YTFhZmVkYTBlZjEwNmE3NjA1NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.m3JeHjUW8Jwl2PEvm22kmpxEgXo9cDUGMd8OZ_uUE0I)
> 
> Consider:
> 
> * The `all` privilege is not required
> * _either_ `monitor` or `manage` is required to check an index
>   
>   * In a future release, `read` may be expanded to be `read` or `read_cross_cluster`, if the Elasticsearch `stats` and `_ilm/explain` APIs are enhanced to support CCS.

This check could look something like:

```ts
export interface DataQualityIndexPrivileges {
    manage: boolean;
    monitor: boolean;
    read: boolean;
    view_index_metadata: boolean;
}

export const hasRequiredPrivileges = ({
  manage,
  monitor,
  read,
  view_index_metadata,
}: DataQualityIndexPrivileges): boolean => (monitor || manage) && read && view_index_metadata;
```

Reply:
Yes, it's implemented here https://github.com/elastic/kibana/pull/175673/files#diff-ab9959d68b85ed70e1f9ea78651355e39de6435aaf05bc28acacac76274c1c30